backend/src/data_classes/Connection.py

```diff
@@ -11,32 +11,39 @@
 
 @dataclass
 class Connection:
-    walking_only: bool
-    start_stop_name: str
-    end_stop_name: str
-    start_datetime: datetime
-    end_datetime: datetime
     transfers: List[IAction]
 
+    @property
+    def walk_only(self):
+        return len(self.transfers) == 1 and isinstance(self.transfers[0], Walk)
+
+    @property
+    def start_stop_name(self):
+        if self.walk_only:
+            return None
+        return self.transfers[0].start_stop_name if isinstance(self.transfers[0], Walk) else self.transfers[1].start_stop_name
```

Comment:
DRY `isinstance(self.transfers[0], Walk)`

use:
```py
@property
def starts_with_walk(self) -> bool:
    return isinstance(self.transfers[0], Walk)
```

analogous for `ends_with_walk`